File: apps/auth-server/src/main/java/com/example/chat/auth/server/api/JwkSetApi.java
package com.example.chat.auth.server.api;

import java.util.Map;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import com.nimbusds.jose.jwk.JWKSet;

import lombok.RequiredArgsConstructor;

/**
 * JWK Set 컨트롤러
 */
@RestController
@RequiredArgsConstructor
public class JwkSetApi {

    private final JWKSet jwkSet;

    /**
     * JWK Set 엔드포인트
     */
    @GetMapping("/.well-known/jwks.json")
    public Map<String, Object> jwkSet() {
        return jwkSet.toJSONObject();
    }
}

File: apps/auth-server/src/main/kotlin/com/example/chat/auth/server/api/JwkSetApi.kt

package com.example.chat.auth.server.api

import com.nimbusds.jose.jwk.JWKSet
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

/**
 * JWK Set 컨트롤러
 */
@RestController
class JwkSetApi(private val jwkSet: JWKSet) {

    /**
     * JWK Set 엔드포인트
     */
    @GetMapping("/.well-known/jwks.json")
    fun jwkSet(): Map<String, Any> {
        return jwkSet.toJSONObject()
    }
}
